src/main/java/org/folio/innreach/domain/service/impl/ContributionValidationServiceImpl.java

```diff
@@ -102,6 +119,19 @@ public boolean isEligibleForContribution(UUID centralServerId, Item item) {
       return false;
     }
 
+    var centralServer = centralServerService.getCentralServer(centralServerId);
+    var itemEffectiveLocation = item.getEffectiveLocationId();
+    var folioLibraryIds = centralServer.getLocalAgencies().stream()
+      .map(LocalAgencyDTO::getFolioLibraryIds)
+      .collect(Collectors.toList());
+    var result = emptyIfNull(folioLibraryIds.stream()
+      .filter(id -> id.contains(itemEffectiveLocation))
+      .collect(Collectors.toList()));
+
+    if (result.isEmpty()) {
+      return false;
+    }
```

Comment:
location is not library. See how it's mapped 
https://github.com/folio-org/mod-inn-reach/blob/b33d7e9f9966975ec8ad51f59abf57f665e85db8/src/main/java/org/folio/innreach/domain/service/impl/RecordTransformationServiceImpl.java#L221

Reply:
thanks for help